sentry-rails/lib/sentry/rails/controller_transaction.rb

```diff
@@ -2,10 +2,34 @@ module Sentry
   module Rails
     module ControllerTransaction
       def self.included(base)
-        base.prepend_before_action do |controller|
-          if Sentry.initialized?
-            Sentry.get_current_scope.set_transaction_name("#{controller.class}##{controller.action_name}", source: :view)
+        base.prepend_around_action(:sentry_around_action)
+      end
+
+      private
+
+      def sentry_around_action
+        if Sentry.initialized?
+          transaction_name = "#{self.class}##{action_name}"
+          Sentry.get_current_scope.set_transaction_name(transaction_name, source: :view)
+          Sentry.with_child_span(op: "view.process_action.action_controller", description: transaction_name) do |child_span|
+            if child_span
+              begin
+                result = yield
+              ensure
+                child_span.set_http_status(response.status)
```

Comment:
should we guard for `response` being nil?

Reply:
In Rails, the controller's request and response object are all [generated at the beginning of the request](https://github.com/rails/rails/blob/847cc9f8cbdaef59fc43efe78ef74eb4b769fa6c/actionpack/lib/action_controller/metal.rb#L272-L273). So in this case, we don't need to worry about it being `nil` 🙂